google-cloud-clients/google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java

```diff
@@ -1042,6 +1050,60 @@ public void testListAllTableData() {
     assertEquals(2, rowCount);
   }
 
+  @Test
+  public void testModelLifecycle() throws InterruptedException {
+
+    String modelName = RemoteBigQueryHelper.generateModelName();
+
+    // Create a model using SQL.
+    String sql =
+        "CREATE MODEL `"
+            + MODEL_DATASET
+            + "."
+            + modelName
+            + "`"
+            + "OPTIONS ( "
+            + "model_type='linear_reg', "
+            + "max_iteration=1, "
+            + "learn_rate=0.4, "
+            + "learn_rate_strategy='constant' "
+            + ") AS ( "
+            + "	SELECT 'a' AS f1, 2.0 AS label "
+            + "UNION ALL "
+            + "SELECT 'b' AS f1, 3.8 AS label "
+            + ")";
+
+    QueryJobConfiguration config = QueryJobConfiguration.newBuilder(sql).build();
+    Job job = bigquery.create(JobInfo.of(JobId.of(), config));
+    job.waitFor();
+    assertNull(job.getStatus().getError());
+
+    // Model is created.  Fetch.
+    ModelId modelId = ModelId.of(MODEL_DATASET, modelName);
+    Model model = bigquery.getModel(modelId);
+    assertNotNull(model);
+    assertEquals(model.getModelType(), "LINEAR_REGRESSION");
+
+    // Mutate metadata.
+    ModelInfo info = model.toBuilder().setDescription("TEST").build();
+    Model afterUpdate = bigquery.update(info);
+    assertEquals(afterUpdate.getDescription(), "TEST");
+
+    // Ensure model is present in listModels.
+    Page<Model> models = bigquery.listModels(MODEL_DATASET);
+    Boolean found = false;
+    for (Model m : models.getValues()) {
+      if (m.getModelId().getModel().equals(modelName)) {
+        found = true;
+        break;
+      }
+    }
+    assertTrue(found);
```

Comment:
You sure you want to assert this since listing resources is eventually consistent? Could be flakey.

Reply:
Good news, everyone!  This should be strongly consistent, per conversation with one of the backend engineers who works on the metadata layer.